woodwork/table_accessor.py

```diff
@@ -228,13 +228,17 @@ def loc(self):
     def schema(self):
         """A copy of the Woodwork typing information for the DataFrame."""
         if self._schema:
-            return self._schema._get_subset_schema(list(self.columns.keys()))
+            return self._schema._get_subset_schema(list(self._dataframe.columns))
 
     @property
     def types(self):
         """DataFrame containing the physical dtypes, logical types and semantic
         tags for the Schema."""
-        return self._schema.types
+        repr_df = self._schema._get_typing_info()
+        # Maintain the same column order used in the DataFrame
+        repr_df = repr_df.loc[list(self._dataframe.columns), :]
+
+        return repr_df
```

Comment:
When we remove `dtype` from the Schema dict in #668 , this'll change again (we'll probably want to maintain collumn order on `WoodworkTableAccessor._get_typing_info`), but this works for the meantime to make sure that the Woodwork table repr follows the same column order as the DataFrame even if changes are made